File: 5Programs/C#/WSMT-5pb/WSMT-5pb/CustomNode.cs
namespace WSMT_5pb
{
    public class CustomNode
    {
        public int Id { get; set; }
        public CustomNode LeftNode { get; set; }
        public CustomNode RightNode { get; set; }
        public string Data { get; set; }
        public CustomNode()
        {

        }
        public CustomNode(int id, CustomNode left, CustomNode right, string data)
        {
            this.Id = id;
            this.LeftNode = left;
            this.RightNode = right;
            this.Data = data;
        }
       
        public override string ToString()
        {
            return "[Id: " + Id + " Data: "+ Data+"]";
        }
    }
}

File: 5Programs/NodeJs/routes/custom_node.js
var method = CustomNode.prototype;

function CustomNode(id, left, right, data) {
    this.id = id;
    this.leftChild = left;
    this.rightChild = right;
    this.data = data;
}

method.getId = function() {
    return this.id;
};
method.setId = function(value) {
   this.id = value;
};
method.getLeftChild = function() {
    return this.leftChild;
};
method.setLeftChild = function(value) {
    this.leftChild = value;
};
method.getRightChild = function() {
    return this.rightChild;
};
method.setRightChild = function(value) {
    this.rightChild = value;
};
method.getData = function() {
    return this.data;
};
method.setData = function(value) {
    this.data = value;
};

method.toString = function(){
    return 'Id ' + this.node_id + ' Data: ' + this.data;
};

module.exports = CustomNode;